docs/sources/reference/components/otelcol/otelcol.receiver.kafka.md

```diff
@@ -253,6 +283,35 @@ If `max_elapsed_time` is set to `"0s"`, failed requests are retried forever unti
 
 `otelcol.receiver.kafka` does not export any fields.
 
+## Supported encodings
+
+`otelcol.receiver.kafka` supports encoding extensions, as well as the following built-in encodings.
+
+Available for all signals:
+
+- `otlp_proto`: the payload is decoded as OTLP Protobuf
+- `otlp_json`: the payload is decoded as OTLP JSON
+
+Available only for traces:
+
+- `jaeger_proto`: the payload is deserialized to a single Jaeger proto `Span`.
+- `jaeger_json`: the payload is deserialized to a single Jaeger JSON Span using `jsonpb`.
+- `zipkin_proto`: the payload is deserialized into a list of Zipkin proto spans.
+- `zipkin_json`: the payload is deserialized into a list of Zipkin V2 JSON spans.
+- `zipkin_thrift`: the payload is deserialized into a list of Zipkin Thrift spans.
+
+Available only for logs:
+
+- `raw`: the payload's bytes are inserted as the body of a log record.
+- `text`: the payload are decoded as text and inserted as the body of a log record. By default, it uses UTF-8 to decode. You can use `text_<ENCODING>`, like `text_utf-8`, `text_shift_jis`, etc., to customize this behavior.
+- `json`: the payload is decoded as JSON and inserted as the body of a log record.
+- `azure_resource_logs`: the payload is converted from Azure Resource Logs format to OTel format.
+
+## Message header propagation
+
+`otelcol.receiver.kafka` will extract Kafka message headers and include them as request metadata (context).
+This metadata can then be used throughout the pipeline, for example to set attributes using `otelcol.processor.attributes`.
```

Comment:
We should have an example of this.

Reply:
Is it really necessary? Even the upstream kafka doc doesn't have such an example. If I do add it, it'd have to look something [like this](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/attributesprocessor/testdata/config.yaml#L344). But I don't feel good about adding an example without testing it, and it'd take me a while to test it.